src/app/services/test-records/test-records.service.ts

```diff
@@ -0,0 +1,45 @@
+import { HttpClient, HttpParams } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { TestResultModel } from '@models/test-result.model';
+import { select, Store } from '@ngrx/store';
+import { fetchTestResults, fetchTestResultsBySystemId, selectAllTestResults, selectedTestResultState, TestResultsState } from '@store/test-records';
+import { Observable, throwError } from 'rxjs';
+import { environment } from '../../../environments/environment';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class TestRecordsService {
+  constructor(private http: HttpClient, private store: Store<TestResultsState>) {}
+
+  fetchTestResultbySystemId(systemId: string, queryparams?: { testResultId?: string }): Observable<Array<TestResultModel>> {
+    if (!systemId) {
+      return throwError(() => new Error('systemId is required'));
+    }
+
+    let params = {};
+    if (queryparams && Object.keys(queryparams).length > 0) {
+      params = new HttpParams({ fromObject: queryparams });
```

Comment:
`queryParams` items should be checked and values verified against security issues and not blindly passed onwards.